### PR TITLE
Consolidated TF modules

### PR DESCRIPTION
**Pull request by eladiw**
#1221 

## What is being addressed

This PR consolidates the most simple modules into the root TF directory.
In order to simplify the dependencies

### What was not changed
The modules for:
appgateway
azure-monitor
network
and resource processor

were not consolidated as each held eith multiple files (e.g. app gateway) or had configurations files in the same directory which makes sense 'bundled' together.

## How is this addressed

- modules which are relatively simple, were moved to the root directory. all inputs/outputs have been adjusted

### CODE REVIEW

File: templates/core/terraform/firewall.tf [hunk 1]
```diff
@@ -1,6 +1,6 @@
 resource "azurerm_public_ip" "fwpip" {
   name                = "pip-fw-${var.tre_id}"
-  resource_group_name = var.resource_group_name
```
Comment (tanya-borisova) [inline templates/core/terraform/firewall.tf:3]: I wouldn't touch firewall too much as it is extracted out in this PR: https://github.com/microsoft/AzureTRE/pull/1226

Comment (eladiw) [reply templates/core/terraform/firewall.tf:3]: Thanks, will remove once merged

**Comment by marrobi** [comment]
@eladiw I'm good with this in principal. Removes a lot of unnecessary TF. However we need to ensure that resources that have state are preserved. 

Thinking about Cosmos and KeyVault particularly which if recreated are going to disrupt a number of existing deployments. The firewall will be moved.

@tanya-borisova is looking into this for the firewall and other shared services.

**Comment by marrobi** [comment]
Another issue is when moving the private endpoint, such as the API app, the IP address changes. App Gateway does not refresh automatically, hence the Web App health status has been in a failed state since the test run. I updated the health probe and came to life. This is a known issue, and will discuss offline.

**Comment by eladiw** [comment]
@marrobi  good points!

Please take a look at the latest revision. In which I've added the 'moved' blocks.
This will allow existing deployments to work with this new version of tf templates.

We can discuss in a few months if we want to delete this file...
https://www.terraform.io/language/modules/develop/refactoring#moved-block-syntax

**Comment by eladiw** [comment]
> @marrobi good points!
> 
> Please take a look at the latest revision. In which I've added the 'moved' blocks. This will allow existing deployments to work with this new version of tf templates.
> 
> We can discuss in a few months if we want to delete this file... https://www.terraform.io/language/modules/develop/refactoring#moved-block-syntax

btw, it was tested using @tamirkamara env-in-pr branch. deployed using 'old' tf, and upgraded without any issues

**Comment by eladiw** [comment]
image

These were not consolidated, as I thought its better to keep them in a separate directory (because they are not as simple). however I am open to discuss this

File: templates/core/terraform/firewall.tf (hunk 1 above)
Comment (eladiw) [reply templates/core/terraform/firewall.tf]: reverted

**Comment by stuartleeks** [comment]
> Please take a look at the latest revision. In which I've added the 'moved' blocks. This will allow existing deployments to work with this new version of tf templates.

Oooh, nice. I hadn't seen that `moved` blocks had been added - that's really useful!

File: templates/core/terraform/bastion.tf [hunk 2]
```diff
@@ -1,6 +1,6 @@
 resource "azurerm_public_ip" "bastion" {
   name                = "pip-bas-${var.tre_id}"
-  resource_group_name = var.resource_group_name
+  resource_group_name = azurerm_resource_group.core.name
   location            = var.location
```
Comment (ross-p-smith) [inline templates/core/terraform/bastion.tf:4]: Shall we do location whilst we are here too? (Take it from the azurerm_resource_group.core.location)

Comment (ross-p-smith) [reply templates/core/terraform/bastion.tf:4]: Absolutely LOVE the "moved" - today I learned something new!

**Review by martinpeck — COMMENTED** [review]
Looks good. A couple of questions: 

- Have you got any numbers/feel for how this improves/changes deployment times?
- Do you think this will cause any issues for those with existing environments? i.e. should we add the `breaking-change` tag to this PR, and then add a note into the docs if there are fixup/warnings?

**Comment by eladiw** [comment]
> Looks good. A couple of questions:
> 
>     * Have you got any numbers/feel for how this improves/changes deployment times?
> 
>     * Do you think this will cause any issues for those with existing environments? i.e. should we add the `breaking-change` tag to this PR, and then add a note into the docs if there are fixup/warnings?

1. unfortunately, I didn't see any improvement regarding the deployment time..
2. The added 'moved' blocks should eliminate a breaking change

**Review by martinpeck — APPROVED** [review]
Thanks @eladiw ! LGTM.

**Comment by martinpeck** [comment]
Please coordinate with @tanya-borisova so that we can merge #1226 first, if possible.

**Review by ross-p-smith — APPROVED** [review]

**Pull request merged**